tests/datasource/test_new_datasource_with_runtime_data_connector.py

```diff
@@ -368,3 +383,343 @@ def test_get_batch_definitions_and_get_batch_basics(
         )
     )
     assert batch.batch_request == {}
+
+
+####################################
+# Tests with data passed in as query
+####################################
+
+
+@pytest.fixture
+def datasource_with_runtime_data_connector_and_sqlalchemy_execution_engine(sa):
+    db_file = file_relative_path(
+        __file__,
+        os.path.join("..", "test_sets", "test_cases_for_sql_data_connector.db"),
+    )
+
+    basic_datasource: Datasource = instantiate_class_from_config(
+        yaml.load(
+            f"""
+    class_name: Datasource
+
+    execution_engine:
+        class_name: SqlAlchemyExecutionEngine
+        connection_string: sqlite:///{db_file}
+
+    data_connectors:
+        test_runtime_data_connector:
+            module_name: great_expectations.datasource.data_connector
+            class_name: RuntimeDataConnector
+            batch_identifiers:
+                - pipeline_stage_name
+                - airflow_run_id
+                - custom_key_0
+        """,
+        ),
+        runtime_environment={"name": "my_datasource"},
+        config_defaults={"module_name": "great_expectations.datasource"},
+    )
+    return basic_datasource
+
+
+def test_datasource_with_runtime_data_connector_and_sqlalchemy_execution_engine_self_check(
+    datasource_with_runtime_data_connector_and_sqlalchemy_execution_engine, sa
+):
+    db_file = file_relative_path(
+        __file__,
+        os.path.join("..", "test_sets", "test_cases_for_sql_data_connector.db"),
+    )
+
+    report = (
+        datasource_with_runtime_data_connector_and_sqlalchemy_execution_engine.self_check()
+    )
+
+    assert report == {
+        "execution_engine": {
+            "connection_string": f"sqlite:///{db_file}",
+            "module_name": "great_expectations.execution_engine.sqlalchemy_execution_engine",
+            "class_name": "SqlAlchemyExecutionEngine",
+        },
+        "data_connectors": {
+            "count": 1,
+            "test_runtime_data_connector": {
+                "class_name": "RuntimeDataConnector",
+                "data_asset_count": 0,
+                "example_data_asset_names": [],
+                "data_assets": {},
+                "note": "RuntimeDataConnector will not have data_asset_names until they are passed in through RuntimeBatchRequest",
+                "unmatched_data_reference_count": 0,
+                "example_unmatched_data_references": [],
+            },
+        },
+    }
+
+
+def test_datasource_with_runtime_data_connector_and_sqlalchemy_execution_engine_error_checking(
+    datasource_with_runtime_data_connector_and_sqlalchemy_execution_engine, sa
+):
+    # interacting with the database using
+    test_query: str = "SELECT * FROM table_full__I;"
```

Comment:
Could we separate out the `pytest.raises` in this one test into multiple tests?

I'm always wary of a lot of asserts in a given test because a single error will throw a failure and prevent all future checks. On the other hand, sometimes it makes a lot of sense to just group similar checks together.

As a happy medium, I think grouping them by "unknown" and "batch_identifiers issues" would be appropriate! What are your thoughts here?

Reply:
You bring up a great point. I'm actually leaning towards splitting it up so that each test only has 1 assert. I'll make the changes to both the query tests and the existing `batch_data` tests before merging  🙇🏼  thanks for the suggestion